compilator/compilator.go

```diff
@@ -336,17 +339,59 @@ func (j compileJob) Run() {
 		stampy.Stamp(c.metricsPath, "fissile", runSeriesName, "start")
 	}
 
-	workerErr := c.compilePackage(c, j.pkg)
-
-	if c.metricsPath != "" {
-		stampy.Stamp(c.metricsPath, "fissile", runSeriesName, "done")
+	exists := false
+	if c.packageStorage != nil {
+		var err error
+		c.ui.Printf("cache: %s %s\n", color.MagentaString("searching for"), j.pkg.Name)
+		exists, err = c.packageStorage.Exists(j.pkg)
+		if err != nil {
+			j.doneCh <- compileResult{pkg: j.pkg, err: err}
+		}
 	}
 
-	c.ui.Printf("done:    %s/%s\n",
-		color.MagentaString(j.pkg.Release.Name),
-		color.MagentaString(j.pkg.Name))
+	// Check to see whether a package already exists in the configured cache
+	// and either download that package or compile and upload it
+	if exists == true {
```

Comment:
This should be just `if exists {`

I was wondering why this wasn't caught by the linter, so decided to run `make` at this time, and realized that the code currently does not pass all the checks from `fmt`, `lint`, etc, so I'll stop now until the project completes `make` successfully.

Make sure you are using the latest versions of Go and the tools. E.g. if you don't see these message from `lint`, then your tools are out-of-date by more than 6 months:

```
==> Linting
/Users/jan/go/src/github.com/SUSE/fissile/compilator/storage.go:53:1: exported function NewPackageStorageFromConfig should have comment or be unexported
/Users/jan/go/src/github.com/SUSE/fissile/compilator/storage.go:58:10: if block ends with a return statement, so drop this else and outdent its block
make: *** [lint] Error 1
```